File: ChallengeBook/CSharp/POJ/Poj1852.cs
#region License

// Copyright (C) 2011-2012 Kazunori Sakamoto
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#endregion

using System;
using System.Collections.Generic;

namespace ChallengeBook.POJ {
	public class Poj1852 {
		public Tuple<int, int> Solve(int length, IEnumerable<int> positions) {
			var minTime = int.MaxValue;
			var maxTime = 0;

			foreach (var position in positions) {
				var min = Math.Min(position, length - position);
				var max = Math.Max(position, length - position);
				minTime = Math.Max(minTime, min);
				maxTime = Math.Max(maxTime, max);
			}

			return Tuple.Create(minTime, maxTime);
		}
	}
}

File: ChallengeBook/Java/src/poj/POJ1852.java
package poj;

import java.io.InputStream;
import java.util.Scanner;

public class POJ1852 {

	public static void main(String[] args) {
		new POJ1852().solve(System.in);
	}

	private void solve(InputStream in) {
		Scanner sc = new Scanner(in);
		int nCases = sc.nextInt();
		for (int i = 0; i < nCases; i++) {
			int length = sc.nextInt();
			int nAnts = sc.nextInt();

			int maxTime = 0, minTime = 0;
			
			for (int j = 0; j < nAnts; j++) {
				int ant = sc.nextInt();
				maxTime = Math.max(maxTime, Math.max(ant, length - ant));
				minTime = Math.max(minTime, Math.min(ant, length - ant));
			}
			
			System.out.println(minTime + " " + maxTime);
		}
	}

}
